src/hardware/sblaster.cpp

```diff
@@ -922,22 +922,62 @@ static std::array<uint8_t, 2> decode_adpcm_4bit(const uint8_t data)
 	        decode_adpcm_portion(data & 0xf, AdjustMap, ScaleMap, LastIndex)};
 }
 
+// Convert sample to float based on type
 template <typename T>
-static const T* maybe_silence(const uint32_t num_samples, const T* buffer)
+static constexpr float to_float(T sample)
+{
+	if constexpr (std::is_same_v<T, uint8_t>) {
+		return lut_u8to16[sample];
+	} else if constexpr (std::is_same_v<T, int8_t>) {
+		return lut_s8to16[sample];
+	} else if constexpr (std::is_same_v<T, uint16_t>) {
+		return static_cast<int16_t>(le16_to_host(sample) -
+		                            Mixer_GetSilentDOSSample<T>());
+	} else if constexpr (std::is_same_v<T, int16_t>) {
+		return static_cast<int16_t>(
+		        le16_to_host(static_cast<uint16_t>(sample)));
+	}
+	static_assert(sizeof(T) <= 2, "Conversion is for 8 & 16-bit ints only!");
+	return 0.0f;
```

Comment:
Sweet Jesus on a stick! 🥶 😆

Reply:
:rofl: I've tried to make them a tad nicer:

```c++
// Convert sample to float based on type
template <typename T>
static constexpr float to_float(T sample)
{
	if constexpr (std::is_same_v<T, uint8_t>) {
		return lut_u8to16[sample];
	}
	if constexpr (std::is_same_v<T, int8_t>) {
		return lut_s8to16[sample];
	}
	if constexpr (std::is_same_v<T, uint16_t>) {
		return static_cast<int16_t>(le16_to_host(sample) -
		                            Mixer_GetSilentDOSSample<T>());
	}
	if constexpr (std::is_same_v<T, int16_t>) {
		return static_cast<int16_t>(
		        le16_to_host(static_cast<uint16_t>(sample)));
	}
	// compile-time checks to prevent the template being misused
	static_assert(std::is_integral_v<T>, "Conversion is only for integers");
	static_assert(sizeof(T) <= 2, "Conversion is only for 8 & 16-bit ints");
	return 0.0f;
```

Reply:
Thanks, that's better 😄